src/GradhSph/GradhSph.cpp

```diff
@@ -167,9 +167,14 @@ void GradhSph<ndim, kernelclass>::DeleteDeadParticles(void)
   // Reorder all arrays following with new order, with dead particles at end
   if (Ndead == 0) return;
 
-  // Reduce particle counters once dead particles have been removed
-  Nhydro -= Ndead;
-  Ntot -= Ndead;
+  // Reduce hydro particle counters once dead particles have been removed and reset all
+  // other particle counters since a ghost and tree rebuild is required.
+  this->NPeriodicGhost = 0;
+  this->Nmpighost      = 0;
+  Nhydro               -= Ndead;
+  Ntot                 = Nhydro;
+
+  // Some sanity checking to ensure there are no dead particles remaining
```

Comment:
Well in theory the rebuild_tree flag should trigger a search for new ghosts. In practice I don't think it happens in all cases.
I think SPH is fine because we never update the ghosts after my changes in the stock_pruned branch. Whereas the meshless will be wrong at the moment (incidentally, we need to do change here also the corresponding function in the meshless), although we discussed that we might move where we do sink accretion anyway.
I'll do the same change to DeleteDeadParticles for the meshless, but I think it should be fixed in the stock_pruned branch and not here.